tests/test_preprints.py

```diff
@@ -462,13 +467,114 @@ def test_format_preprint(self):
 
         assert nodes == {}
 
+    def test_format_thesis(self):
```

Comment:
I'll do my best. The fact is I just copied the test from ```test_format_preprint``` and substituted ```thesis``` where appropriate. It is currently using the same ```Graph``` template. So I'll try to make comments, but I'll have to figure out what it is doing first.